### PR TITLE
Make client events compatible with ibc-go v5.0.0

### LINKED ISSUE #144: Make client events compatible with ibc-go v5.0.0

### PR DESCRIPTION
**Pull request by plafer**
Closes: #144

## Description

Left to do:
- [x] client events are compatible with ibc-go
- [x] ~~Handlers also [emit the `EventTypeMessage` event](https://github.com/cosmos/ibc-go/blob/2056c2d73c56fec592a619ab26123d59215198c3/modules/core/02-client/keeper/events.go#L26-L30)~~ (this is SDK-specific)

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests.
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/ibc/src/events.rs [hunk 1]
```diff
@@ -153,13 +149,11 @@ pub enum IbcEventType {
     TimeoutOnClose,
     AppModule,
     Empty,
```
Comment (hu55a1n1) [inline crates/ibc/src/events.rs]: I think we can also remove `IbcEventType::Empty` here.

File: crates/ibc/src/core/ics02_client/events.rs [hunk 2]
```diff
@@ -21,262 +21,324 @@ pub const CLIENT_TYPE_ATTRIBUTE_KEY: &str = "client_type";
...
+        Tag {
+            key: CLIENT_ID_ATTRIBUTE_KEY.parse().unwrap(),
+            value: attr.client_id.to_string().parse().unwrap(),
+        }
     }
-    pub fn set_height(&mut self, height: Height) {
-        self.height = height;
+}
+
+impl From<ClientId> for ClientIdAttribute {
```
Comment (hu55a1n1) [inline crates/ibc/src/core/ics02_client/events.rs]: I would suggest using `derive_more::From` to derive these `From` implementations for all attributes.

File: crates/ibc/src/core/ics02_client/events.rs [hunk 3]
```diff
@@ -21,262 +21,324 @@ pub const CLIENT_TYPE_ATTRIBUTE_KEY: &str = "client_type";
...
-        CreateClient(attrs)
+    pub fn consensus_height(&self) -> &Height {
+        &self.consensus_height.consensus_height
     }
 }
 
-impl From<CreateClient> for IbcEvent {
-    fn from(v: CreateClient) -> Self {
-        IbcEvent::CreateClient(v)
+impl Display for CreateClient {
```
Comment (hu55a1n1) [inline crates/ibc/src/core/ics02_client/events.rs]: Noticed that the `Display` implementation has changed. Not sure if we still need a `Display` implementation - might we be able to remove the `IbcEvent`'s `Display` implementation altogether?

File: crates/ibc/src/core/ics02_client/events.rs [hunk 4]
```diff
@@ -21,262 +21,324 @@ pub const CLIENT_TYPE_ATTRIBUTE_KEY: &str = "client_type";
...
+    client_id: ClientId,
 }
 
-impl NewBlock {
-    pub fn new(h: Height) -> NewBlock {
-        NewBlock { height: h }
+impl From<ClientIdAttribute> for Tag {
+    fn from(attr: ClientIdAttribute) -> Self {
+        Tag {
+            key: CLIENT_ID_ATTRIBUTE_KEY.parse().unwrap(),
```
Comment (hu55a1n1) [inline crates/ibc/src/core/ics02_client/events.rs:37]: Would be nice to have tests that verify the `AbciEvent` keys match the events actually emitted by ibc-go. (We could do this in a later PR.)

**Review by hu55a1n1 — APPROVED** [review]
Great stuff! Preapproved with some minor suggestions.

File: crates/ibc/src/events.rs (hunk 1 above)
Comment (plafer) [reply crates/ibc/src/events.rs]: [2280d1f](https://github.com/cosmos/ibc-rs/pull/138/commits/2280d1f9844f1c3c53b21ae86fd2b6ba238e38c2)

File: crates/ibc/src/core/ics02_client/events.rs (hunk 2 above)
Comment (plafer) [reply crates/ibc/src/core/ics02_client/events.rs]: Good idea! [bc9274a](https://github.com/cosmos/ibc-rs/pull/138/commits/bc9274a6fa2f46ecdda60abe4bfa412603c9423b)

File: crates/ibc/src/core/ics02_client/events.rs (hunk 3 above)
Comment (plafer) [reply crates/ibc/src/core/ics02_client/events.rs]: Agreed [31301ab](https://github.com/cosmos/ibc-rs/pull/138/commits/31301ab18488130856c8fe920a262df0023dd684)

File: crates/ibc/src/core/ics02_client/events.rs (hunk 4 above)
Comment (plafer) [reply crates/ibc/src/core/ics02_client/events.rs:37]: #163

**Pull request merged**